### PR TITLE
PUB 1045 - Metadata Endpoint

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1045

### Change description ###

Updated the /artefact/<artefactId> endpoint, to return the Artefact Metadata (The same as the POST endpoint), rather than the payload.

Added /artefact/<artefactId>/payload, which will return the payload of the artefact.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java [hunk 1]
```diff
@@ -202,18 +202,35 @@ public PublicationController(PublicationService publicationService, ValidationSe
...
     }
 
+    @ApiResponses({
+        @ApiResponse(code = 200,
+            message = "Gets the artefact metadata",
+            response = Artefact.class),
+        @ApiResponse(code = 404,
+            message = "No artefact found matching given parameters and date requirements"),
+    })
+    @ApiOperation("GGets the metadata for the blob, given a specific artefact id")
```
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java:212]: youve double G'd 😏

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java [hunk 2]
```diff
@@ -202,18 +202,35 @@ public PublicationController(PublicationService publicationService, ValidationSe
...
             response = String.class),
         @ApiResponse(code = 404,
             message = "No artefact found matching given parameters and date requirements"),
     })
-    @ApiOperation("Get the info from within a blob given source artefact id and provenance")
-    @GetMapping("/{artefactId}")
-    public ResponseEntity<String> getBlobData(@PathVariable UUID artefactId, @RequestHeader Boolean verification) {
+    @ApiOperation("Gets the the payload for the blob, given a specific artefact ID")
+    @GetMapping("/{artefactId}/payload")
+    public ResponseEntity<String> getArtefactPayload(
```
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java:230]: does this need to be the endpoint that returns string if its json but MultipartFile if its a flat file?

Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java:230]: or is that a diff ticket

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java (hunk 1 above)
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java]: Good spot! Removed :)

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java (hunk 2 above)
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java:231]: My vote would be a different ticket. I did have a quick look to see if it was a quick thing I could include as part of this, however I think it requires a think about extensions, file naming so thinking we leave it for now

**Review by furnivall — DISMISSED** [review]
👍

**Review by furnivall — DISMISSED** [review]
👍

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java (hunk 2 above)
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/PublicationController.java:231]: 👍🏼  we'll need to make sure thats a captured ticket, could be able to extend this method to return either type when we get round to it

**Review by kurt-uk — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
👍

**Pull request merged**